### DISCUSSION TITLE
I get an error when starting the workspace:Error: Failed to install provider.   Why is that?

### DISCUSSION THREAD

**Discussion by simRay56** [coder/coder#15760] [Workspace Provisioning & Templates]
Initializing the backend...
Initializing provider plugins...
- Finding coder/coder versions matching "~> 0.7.0"...
- Finding kreuzwerker/docker versions matching "~> 3.0.1"...
- Installing coder/coder v0.7.0...
- Installing kreuzwerker/docker v3.0.2...
Error: Failed to install provider
Error while installing coder/coder v0.7.0: failed to open temporary file to
download from
https://github.com/coder/terraform-provider-coder/releases/download/v0.7.0/terraform-provider-coder_0.7.0_linux_amd64.zip:
open /tmp/terraform-provider1185905871: no such file or directory
Error: Failed to install provider
Error while installing kreuzwerker/docker v3.0.2: failed to open temporary
file to download from
https://github.com/kreuzwerker/terraform-provider-docker/releases/download/v3.0.2/terraform-provider-docker_3.0.2_linux_amd64.zip:
open /tmp/terraform-provider421569086: no such file or directory

**Comment by ethanndickson** [comment]
It sounds like `terraform` is unable to create a temporary directory to download the providers into.

From https://stackoverflow.com/questions/74696348/terraform-init-error-failed-to-open-temporary-file-with-direnv

> Terraform (and any go program that calls [os.TempDir("", …)](https://pkg.go.dev/os#TempDir) or [os.MkdirTemp("", …)](https://pkg.go.dev/os#MkdirTemp), or [ioutil.TempFile("", …)](https://pkg.go.dev/io/ioutil#TempFile), or [ioutil.TempDir("", …)](https://pkg.go.dev/io/ioutil#TempDir)) creates files in the directory returned by [os.TempDir()](https://pkg.go.dev/os#TempDir), which is the value of the environment variable `$TMPDIR`. `$TMPDIR` must exist first, or else those functions give an error.

Can you verify that `coder server` (and it's provisioners) has permission to create directories in `$TMPDIR`?

**Comment by simRay56** [comment]
> It sounds like `terraform` is unable to create a temporary directory to download the providers into.
> 
> From https://stackoverflow.com/questions/74696348/terraform-init-error-failed-to-open-temporary-file-with-direnv
> 
> > Terraform (and any go program that calls [os.TempDir("", …)](https://pkg.go.dev/os#TempDir) or [os.MkdirTemp("", …)](https://pkg.go.dev/os#MkdirTemp), or [ioutil.TempFile("", …)](https://pkg.go.dev/io/ioutil#TempFile), or [ioutil.TempDir("", …)](https://pkg.go.dev/io/ioutil#TempDir)) creates files in the directory returned by [os.TempDir()](https://pkg.go.dev/os#TempDir), which is the value of the environment variable `$TMPDIR`. `$TMPDIR` must exist first, or else those functions give an error.
> 
> Can you verify that `coder server` (and it's provisioners) has permission to create directories in `$TMPDIR`?

Hi!
I've seen my $TMPDIR, and the permissions are given to drwxrwxrwt
It used to be possible to build a workspace, but recently it has suddenly stopped working, even for worksapce.

**Comment by matifali** [comment]
@yangxup can you manually access https://github.com/coder/terraform-provider-coder/releases/download/v0.7.0/terraform-provider-coder_0.7.0_linux_amd64.zip from your network? 
It could be an issue where your machine is not bale to access Github.

**Reply by simRay56** [reply]
@matifali My network has access to github, I tried to download these packages to the specified folder on the server, but the build template still gave an error.

**Reply by matifali** [reply]
@yangxup are you still stuck on this? Is this consistently reproducible?
